odoo/addons/base/models/ir_module.py

```diff
@@ -563,6 +564,14 @@ def next(self):
         }
 
     def _button_immediate_function(self, function):
+        if getattr(threading.currentThread(), 'testing', False):
+            _logger.warning(
+                "Module operations inside tests are not transactional and this forbidden.\n"
+                "If you really need to perform module operations to test a specific behavior, it "
+                "is best to write it as a standalone script, and ask the runbot/metastorm team "
+                "for help."
+            )
+            return False
```

Comment:
For the sake of consistency, either the method does what it is supposed to do, or it crashes.  If you log this as a warning, the method should behave normally.  If you really want to forbid it, raise an exception.

Reply:
I didn't know which kind of error would be appropriate here, I decided to go with `RuntimeError` for now, I don't think that there's an existing Error type that could cover it.